src/renderer.tsx

```diff
@@ -44,4 +59,145 @@ export default function (context): void {
 		statuses.push(BackupStates.creatingDatabaseSnapshot);
 		return statuses;
 	});
+
+	// add new routes and components to Local core
+	hooks.addFilter('AddSiteIndexJS:RoutesArray', (routes, path) => {
+		const cloudBackupRoutes = [
+			{ key: 'add-site-choose', path: `${path}/`, component: ChooseCreateSiteHOC },
+			{ key: 'add-site-select-site-backup', path: LOCAL_ROUTES.ADD_SITE_BACKUP_SITE, component: SelectSiteBackupHOC },
+			{ key: 'add-site-select-snapshot', path: LOCAL_ROUTES.ADD_SITE_BACKUP_SNAPSHOT, component: SelectSnapshotHOC },
+		];
+
+		routes.forEach((route) => {
+			if (route.path === `${path}/`) {
+				cloudBackupRoutes.push(
+					{ key: route.key, path: LOCAL_ROUTES.ADD_SITE_CREATE_NEW, component: route.component },
+				);
+			}
+			cloudBackupRoutes.push(route);
+		});
+
+		return cloudBackupRoutes;
+	});
+
+	// optionally modify NewSiteEnvironment component functionality in Local core
+	hooks.addFilter('AddSiteIndexJS:NewSiteEnvironment', (newSiteEnvironmentProps) => {
+		if (newSiteEnvironmentProps.siteSettings.cloudBackupMeta?.createdFromCloudBackup) {
+			const continueCreateSite = () => {
+				LocalRenderer.sendIPCEvent('addSite', {
+					newSiteInfo: newSiteEnvironmentProps.siteSettings,
+					goToSite: true,
+					installWP: false,
+				});
+			};
+
+			const onGoBack = () => {
+				LocalRenderer.sendIPCEvent('goToRoute', LOCAL_ROUTES.ADD_SITE_BACKUP_SNAPSHOT);
+			};
+
+			return {
+				...newSiteEnvironmentProps,
+				onContinue: continueCreateSite,
+				onGoBack: onGoBack,
```

Comment:
Can you use the shorthand here? So just `onGoBack` instead of `onGoBack: onGoBack`